Migrate to prek

ulgens opened the pull request:
Similar to:
- https://github.com/prettytable/prettytable/pull/412

Tested this on the fork, it's working fine: https://github.com/ulgens/django-commons-packages-summary/pull/1/checks

ulgens committed:
Migrate to prek

ulgens commented:
@hugovk With `lint.yml` in place, is this integration necessary?

<img width="806" height="94" alt="Screenshot 2025-11-27 at 16 47 46 UTC@2x" src="https://github.com/user-attachments/assets/166cab5b-e9c0-4b3b-9b5e-c82663da2f90" />

hugovk commented on .github/workflows/lint.yml:
```diff
@@ -4,7 +4,6 @@ on: [push, pull_request, workflow_dispatch]
 
 env:
   FORCE_COLOR: 1
```
```suggestion
  FORCE_COLOR: 1
  RUFF_OUTPUT_FORMAT: github
```

ulgens replied:
Resolved.

hugovk commented on .github/workflows/lint.yml:
```diff
@@ -4,7 +4,6 @@ on: [push, pull_request, workflow_dispatch]
 
```
```suggestion

permissions: {}

```

ulgens replied:
Resolved.

hugovk replied:
Please could you also move it to here? Then it'll match my other repos' files and be easier to diff :)

hugovk commented on .github/workflows/lint.yml:
```diff
@@ -15,8 +14,4 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
```
```suggestion
      - uses: actions/checkout@v4
        with:
          persist-credentials: false
```

ulgens replied:
Resolved.

ulgens replied:
Asking to learn, do you have any references about why this is a bad thing? I found https://github.com/actions/checkout/issues/485 with a quick search but I feel like there is more to learn about this issue.

hugovk replied:
It's one of the things zizmor flags: https://docs.zizmor.sh/audits/#artipacked

ulgens replied:
Thank you.

hugovk commented on .github/workflows/lint.yml:
```diff
@@ -15,8 +14,4 @@ jobs:
-          python-version: "3.x"
-          cache: pip
-      - uses: pre-commit/action@v3.0.1
+      - uses: j178/prek-action@v1.0.11
```
```suggestion
      - uses: j178/prek-action@v1
```

ulgens replied:
Resolved.

hugovk reviewed (commented):
Thanks! A few other suggestions.

> @hugovk With `lint.yml` in place, is this integration necessary?

Yes, it will autoupdate PRs, and also does the quarterly update.

hugovk commented:
btw I realised there was no licence here so just added MIT:

https://github.com/hugovk/hugovk/blob/main/LICENCE.md

ulgens commented:
> Yes, it will autoupdate PRs, and also does the quarterly update.

In the fork I plan to maintain, my plan was to migrate the project to `pyproject.toml` (and uv) and enable Renovate for all dependency updates (including pre-commit hooks) I can push it to upstream if that sounds useful.

ulgens committed:
Update ruff output format

https://docs.astral.sh/ruff/settings/#output-format
https://github.com/hugovk/hugovk/pull/15#discussion_r2569126111

ulgens committed:
Revoke permissions

https://github.com/hugovk/hugovk/pull/15#discussion_r2569128164
https://github.com/hugovk/hugovk/pull/15#discussion_r2571674642
https://docs.zizmor.sh/audits/#artipacked

hugovk commented on .github/workflows/lint.yml:
```diff
@@ -4,19 +4,17 @@ on: [push, pull_request, workflow_dispatch]
-          cache: pip
-      - uses: pre-commit/action@v3.0.1
+          persist-credentials: false
+
```
```suggestion
```

hugovk replied:
Tip for next time: you can use commit these suggestions directly via the GH UI. It's easier for you than having to edit files, and easier for me to check they were applied :)

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

ulgens replied:
Got it. I was trying to keep the commit history clean.

hugovk commented:
Thanks!
